### DISCUSSION TITLE
How can I integrate already-created (LaTeX) pdf's in a Quarto website?

### DISCUSSION THREAD

**Discussion by dense-set** [quarto-dev/quarto-cli#5863] [Q&A] [answered]
##### [I have also asked the exact same question on StackOverflow, so please don't get mad.]

I am making a website with Quarto (to be hosted on GitHub Pages) that documents my already finished LaTeX projects. A project is a directory which contains the following:
- A .sty file
- A master and several `\include`d .tex files.
- The rendered pdf.
- The usual auto-generated files.

I wish to include the pdf's of these projects to my website as pages (like each `.qmd` creates a page) **as they are** (with no addition formatting due to Quarto).

I searched the [Quarto Guide][1] but it unfortunately does not provide any information on how to have already-created pdf's on the website.

  [1]: https://quarto.org/docs/websites/

**Comment by mcanouil** [comment, marked as answer]
To include almost anything in HTML, `iframe` is the solution.

See an example:

- HTML page: <https://www.andrewheiss.com/cv/>
- Source code: <https://github.com/andrewheiss/ath-quarto/blob/main/cv/index.qmd>

**Reply by dense-set** [reply]
Thanks, but this doesn't seem very elegant. Is there a way to make them appear like what a .qmd file will render with `format: pdf`, the way a typical lecture notes pdf opens online (e.g. [this](http://alpha.math.uga.edu/~pete/factorization2010.pdf))?

**Reply by mcanouil** [reply]
I don't understand, _i.e._, a PDF is not and will never be an HTML page.

The solution from above is how to embed/include the PDF into an HTML page.

If you want it to be open in the browser (as long as it has the plugins to open PDF), then setup a link to the file and add that file into your project yaml.
Note that, when opening the PDF this way, you'll leave the website.

```yaml
project:
  resources:
    - path/to/file.pdf
```

**Reply by pglpm** [reply]
The iframe is a neat solution. Good to point out that whether the pdf gets actually visualized within the browser, or not, depends on the settings chosen by its user. For example I [disabled Firefox's built-in viewer](https://support.mozilla.org/en-US/questions/943119) so I only see a download button.

For this same reason the link "(e.g. this)" that @dense-set  encloses, also appears to me like a dialogue to save the file – I don't see any pdf in the browser.

**Reply by dense-set** [reply]
Ah, I now understand what you mean. I have never worked with HTML or webdev before, and naïvely assumed that those "(e.g. this)" pdf's also were webpages. @pglpm, your comment has made it clear now.

So I will mark @mcanouil's ifframe solution as the answer. 

Just one last thing: I have added my `file.pdf` to the `resources` and using `![](file.pdf)` in a .qmd file creates a small subwindow inside the webpage where I can view the pdf. How to instead create a link in the webpage that opens the pdf in the browser as you say, @mcanouil?

**Reply by pglpm** [reply]
@mcanouil 's `iframe` solution is neat because not only your pdf will be visible in the web page (if allowed by the user), but you can also have extra HTML navigation elements, say a header or similar. For example, now I opened in the browser that "[Factorization in integral domains](http://alpha.math.uga.edu/~pete/factorization2010.pdf)" pdf-page, but I don't know where to click to go to the author's main web page, if there is one. With an iframe, such a link could have been available.
Of course the decision is purely yours :) I'm just commenting to weigh the options and their results.

**Reply by dense-set** [reply]
More the options, more the headache. :') 
jk, thanks.

**Comment by joelostblom** [comment]
You can also do this by manually (e.g. as part of a Makefile) copy the PDFs the the built location. E.g. I'm creating a book output that  consists of `ipynb` files. In a cell in these files, I have something like:

```html
<html>
  <body>
    <iframe src="slides-lec2.pdf#zoom=80&&navpanes=0&statusbar=0&messages=0&pagemode=none" width="100%" height="475px" style="position: relative;z-index: 9999999;">
    </iframe>
  </body>
```

Then I manually need to copy the pdf `slides-lec2.pdf` into the same location as the `ipynb` output file (e.g. `_build/_book/lectures`). I think quarto should be able to do the copying, but when I tried to reference the pdf in the metadata yaml to have quarto detect it, I couldn't get it to work and I can't find anything in the docs about how to do this.

I also noticed that someone made an extension https://github.com/jmgirard/embedpdf/blob/main/README.md (I haven't tried it)
</html>

**Reply by joelostblom** [reply]
For including PDF (or other files) automatically, something like this can be included in the project's `_quarto.yml` folder:

```yaml
project:
  type: book
  resources:
    - slides-lec[1-8].pdf
```

It is only listed under [the manuscript section in the docs,](https://quarto.org/docs/manuscripts/components.html#including-other-resources) but works with other outputs.

**Reply by mcanouil** [reply]
There is no need to make it that complicated, use markdown and Quarto should discover the resource properly.

```md
![](slides123.pdf)
```

If Quarto does not discover the resource, then you can use `resources` as mentioned nearly at the beginning of this thread: https://github.com/quarto-dev/quarto-cli/discussions/5863#discussioncomment-6132641

**Reply by mcanouil** [reply]
FYI, a related discussion:

- https://github.com/quarto-dev/quarto-cli/discussions/7918

**Reply by joelostblom** [reply]
Thanks for the reply! I didn't think that would work with the PDF options like zoom level, but it seems like I can just append them like this, which is convenient:

```
![](slides-lec2.pdf#zoom=90&&navpanes=0&statusbar=0&messages=0&pagemode=none){width=100% height=475px}
```

> ...use markdown and Quarto should discover the resource properly.

Unfortunately this does not happen. I have to declare the resources specifically in the YAML file as per my post above.
